components/LogWorkout/LogWorkout.jsx

```diff
@@ -23,18 +23,22 @@ const TitleText = styled.Text`
   margin: 15px 15px;
 `;
 
-const parseExercises = (exercises) => exercises.map((exercise) => ({
-  name: exercise.name,
-  color: exercise.color,
-  type: exercise.type,
-  sets: exercise.sets.map((set) => ({
-    prevWeight: set.weight,
-    weight: '',
-    prevDuration: set.duration,
-    duration: '',
-    completed: false,
-  })),
-}));
+const parseExercises = (exercises) => exercises.map((exercise) => {
+  const exerciseType = exercise.sets[0].reps ? 'REPS' : 'SECS';
+
+  return {
+    name: exercise.name,
+    color: exercise.color,
+    type: exerciseType,
+    sets: exercise.sets.map((set) => ({
+      prevWeight: set.weight && set.weight.toString(),
+      weight: '',
+      prevDuration: exerciseType === 'REPS' ? set.reps.toString() : set.time.toString(),
```

Comment:
The set.reps.toString() and set.time.toString() will cause an error if they are left blank

Reply:
Yeah, they should never both be undefined though. They will be for newly created workouts but once AddWorkout is updated to reflect the new workout structure it should never occur